hadoop-plugin/src/main/groovy/com/linkedin/gradle/azkaban/yaml/YamlCompiler.groovy

```diff
@@ -0,0 +1,50 @@
+package com.linkedin.gradle.azkaban.yaml;
+
+import com.linkedin.gradle.hadoopdsl.NamedScope;
+import com.linkedin.gradle.hadoopdsl.Workflow;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Turns the internal representations of workflows, jobs, and properties into their
+ * corresponding Yaml counterparts then creates and fills the <workflow-name>.flow and
+ * <project-name>.project yaml files.
+ *
+ * The usage of Yaml compilation vs .job/.properties compilation is user-controlled within their
+ * workflows.gradle file (or equivalent). The default is .job/.properties for now.
+ */
+class YamlCompiler {
+  Yaml yaml;
+  YamlWorkflow yamlWorkflow;
+  YamlProject yamlProject;
+
+  /**
+   * Construct YamlCompiler from Workflow.
+   * Expects that buildFlowTargets method has already been executed.
+   *
+   * @param workflow The built workflow to be converted into Yaml and output
+   * @param parentScope The parent scope of the workflow
+   */
+  YamlCompiler(Workflow workflow, NamedScope parentScope) {
+    DumperOptions options = new DumperOptions();
+    options.setDefaultFlowStyle(DumperOptions.FlowStyle.BLOCK);
+    yaml = new Yaml(options);
+    yamlWorkflow = new YamlWorkflow(workflow, parentScope, false);
+    yamlProject = new YamlProject();
+  }
+
+  /**
+   * Write the YamlWorkflow and YamlProject yaml files into the proper directory
+   *
+   * @param parentDirectory The directory where the Yaml Workflow will be written to
+   * @param workflowName The name of the workflow being output
+   */
+  void dumpYaml(String parentDirectory, String workflowName, String projectName) {
+    def initYamlFile = { fileName ->
+      return new FileWriter(new File(parentDirectory, fileName));
```

Comment:
Do we need to close the writer after done?

Reply:
Yes - I think we should close the writer. I'll tweak the PR.

In terms of handling the IOException - Groovy doesn't force us to. If the exception is thrown then the execution would fail and a related IOException would be thrown.

It would be nice to attach additional information to this IOException - but no logger is present at this time. Should I just write to stdout? I imagine that would be appropriate for the hadoopDSL.